core/services/flux_monitor.go

```diff
@@ -206,24 +209,28 @@ func (fm *concreteFluxMonitor) RemoveJob(ID *models.ID) {
 // DeviationCheckerFactory holds the New method needed to create a new instance
 // of a DeviationChecker.
 type DeviationCheckerFactory interface {
-	New(models.Initiator, RunManager) (DeviationChecker, error)
+	New(models.Initiator, RunManager, *orm.ORM) (DeviationChecker, error)
 }
 
 type pollingDeviationCheckerFactory struct{}
 
-func (f pollingDeviationCheckerFactory) New(initr models.Initiator, runManager RunManager) (DeviationChecker, error) {
+func (f pollingDeviationCheckerFactory) New(initr models.Initiator, runManager RunManager, orm *orm.ORM) (DeviationChecker, error) {
 	if initr.InitiatorParams.PollingInterval < MinimumPollingInterval {
 		return nil, fmt.Errorf(
 			"pollingInterval must be equal or greater than %s",
 			MinimumPollingInterval,
 		)
 	}
 
+	urls, err := ExtractFeedURLs(initr.InitiatorParams, orm)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
@se3000 @RyanRHall what are your thoughts on declaring functions before they are used?  It may be better if readers were introduced to the function before its used, and people generally read code from top to bottom.

Reply:
I think you and @HenryNguyen5 have opposite POVs here :laughing: I think he introduced me to the idea that you start with the most important functions and scroll down for granularity. Do we have any kind of developer style guide for things like this?

Reply:
No its pretty much ad-hoc, just food for thought really.  I'm unaware of anything established.

I generally find myself jumping to the function to see what's happening.  For instance, the type signature of the function isn't always obvious in the name.

Reply:
Yeah, I dont think there's any styleguide. I tend to write functions that are composed of other functions at the top of the file like done here, so you can get context of how and why they're used before you dive into the weeds and look at the building blocks used

Reply:
My understanding for standard Go style is to group functions and methods for a type together. Most important type goes at the top of the file, and related things below it, descending in priority. I can't say that we always follow that. I'm pretty sure that enforcing methods need to be declared before they're used would end up with a lot of cyclical problems.